src/include/gurt/common.h

```diff
@@ -204,12 +204,50 @@ extern "C" {
 #define D_ALLOC_ARRAY(ptr, count) D_ALLOC_CORE(ptr, sizeof(*ptr), count)
 #define D_FREE_PTR(ptr)		D_FREE(ptr)
 
-#define D_GOTO(label, rc)			\
-	do {					\
-		__typeof__(rc) __rc = (rc);		\
-		(void)(__rc);			\
-		goto label;			\
+/* _Generic() is part of the c11 standard but isn't yet supported by
+ * gcc until version 4.9 so check for that but enable it in all other
+ * cases.
+ */
+
+#ifdef __GNU__
+#if __GNUC__ > 4 || \
```

Comment:
So it's only available for gcc < 4.9 ?   Or should this be < 4 and <= 9 ?  Also, what is the extra || on 214 ?

Reply:
It's not in 4.8 or earlier.